platform-sdk/swirlds-config-impl/src/main/java/com/swirlds/config/impl/internal/ConfigurationBuilderImpl.java

```diff
@@ -185,4 +186,18 @@ private <T extends Record> void addConfigDataType(@NonNull final Class<T> type)
         }
         configuration.addConfigDataType(type);
     }
+
+    /**
+     * Sets the value for the config.
+     *
+     * @param propertyName name of the property
+     * @param value        the value
+     * @return the {@link ConfigurationBuilder} instance (for fluent API)
+     */
+    @NonNull
+    public ConfigurationBuilder withValue(@NonNull final String propertyName, @NonNull final Object value) {
+        Objects.requireNonNull(propertyName, "propertyName must not be null");
+        Objects.requireNonNull(value, "value must not be null");
+        return withSource(new SimpleConfigSource(propertyName, value.toString()));
```

Comment:
That implementation is not good from my point of view. If you provide multiple values it would end in multiple sources that all have the same ordinal. That will be hard to handle. Please store a map in the impl class and create a single SimpleConfigSource with very low ordinal once build() is called.

Reply:
I didn't realize that was possible, I was following the pattern in `TestConfigBuilder.java`, which creates a new source each time. I agree that the way you suggest is a lot better. Change made.